src/Instrumentation/AwsSdk/AwsSdkInstrumentation.php

```diff
@@ -37,8 +46,90 @@ public function init(): bool
         return true;
     }
 
+    public function setPropagator(TextMapPropagatorInterface $propagator): void
+    {
+        $this->propagator = $propagator;
+    }
+
+    public function getPropagator(): TextMapPropagatorInterface
+    {
+        return $this->propagator;
+    }
+
+    public function setTracerProvider(TracerProviderInterface $tracerProvider): void
+    {
+        $this->tracerProvider = $tracerProvider;
+    }
+
+    public function getTracerProvider(): TracerProviderInterface
+    {
+        return $this->tracerProvider;
+    }
+
+    public function getTracer(): TracerInterface
+    {
+        return $this->tracerProvider->getTracer('io.opentelemetry.contrib.php');
+    }
+
+    public function endRootSpan()
+    {
+        $this->root->end();
+    }
+
     public function activate(): bool
     {
+        AwsGlobal::setInstrumentation($this);
+
+        try {
+            $this->root = $this->getTracer()->spanBuilder('AwsSDKInstrumentation')->setSpanKind(SpanKind::KIND_CLIENT)->startSpan();
+            $this->root->activate();
```

Comment:
very important that you store the `$scope` you get back from activating a span, and ensure you `::detach()` it when you end the root span. This goes for any span that you activate, and we recommend doing it in a try/finally block